src/ds/logger.h

```diff
@@ -94,17 +94,26 @@ namespace logger
       const std::tm& host_tm,
       const ::timespec& host_ts,
       uint16_t thread_id,
-      const std::optional<::timespec>& enclave_ts = std::nullopt) override
+      const std::optional<float>& enclave_offset = std::nullopt) override
     {
       nlohmann::json j;
       j["m"] = msg;
 
-      if (enclave_ts.has_value())
+      if (enclave_offset.has_value())
       {
+        ::timespec enc_ts = host_ts;
+        enc_ts.tv_sec += (size_t)enclave_offset.value();
+        enc_ts.tv_nsec +=
+          (long long)(enclave_offset.value() * ns_per_s) % ns_per_s;
+
+        if (enc_ts.tv_nsec > ns_per_s)
+        {
+          enc_ts.tv_sec += 1;
+          enc_ts.tv_nsec -= ns_per_s;
+        }
+
         std::tm enclave_tm;
-        auto enc_ts = enclave_ts.value();
-        ::timespec_get(&enc_ts, TIME_UTC);
-        ::gmtime_r(&enc_ts.tv_sec, &enclave_tm);
```

Comment:
This was just wrong. In the JSON logging path, we ignored the passed enclave/offset time, and just re-fetched the host time again!

Reply:
Previously:
```
{"h_ts":"2021-04-26T10:26:02.030865Z","e_ts":"2021-04-26T10:26:02.030865Z",
{"h_ts":"2021-04-26T10:26:03.402864Z","e_ts":"2021-04-26T10:26:03.402865Z",
```
Identical times!

Now:
```
{"h_ts":"2021-04-26T10:27:12.872847Z","e_ts":"2021-04-26T10:27:12.871473Z",
{"h_ts":"2021-04-26T10:27:13.094136Z","e_ts":"2021-04-26T10:27:13.092686Z",
```
Enclave timestamp is slightly behind host timestamp, as in non-JSON logging.